src/providers/azure/mod.rs

```diff
@@ -216,7 +217,15 @@ impl Azure {
         // make sure the metadata service is compatible with our version
         azure
             .is_fabric_compatible(MS_VERSION)
-            .chain_err(|| "failed version compatibility check")?;
+            .map_err(|_| {
+                let user = Uid::current();
+                let root_or_not = Uid::is_root(user);
+                if !root_or_not {
+                    warn!("unable to reach Azure endpoints, please check whether firewall rules are blocking access to them");
```

Comment:
It would be good to have a short comment on top of this to mention why we specifically care about root, and a link to https://github.com/coreos/bugs/issues/2468.